ground/src/main/java/com/google/android/ground/repository/SubmissionRepository.kt

```diff
@@ -143,37 +144,49 @@ class SubmissionRepository @Inject constructor(
             }
     }
 
-    fun deleteSubmission(submission: Submission): @Cold Completable =
-        applyAndEnqueue(
-            builder()
-                .setJob(submission.job)
-                .setSubmissionId(submission.id)
-                .setResponseDeltas(ImmutableList.of())
-                .setType(Mutation.Type.DELETE)
-                .setSyncStatus(SyncStatus.PENDING)
-                .setSurveyId(submission.survey.id)
-                .setLocationOfInterestId(submission.locationOfInterest.id)
-                .setClientTimestamp(Date())
-                .setUserId(authManager.currentUser.id)
-                .build()
-        )
+    fun deleteSubmission(submission: Submission): @Cold Completable {
+        return if (submission.locationOfInterest.id == null) {
```

Comment:
Nit: Imo short-circuiting is still more readable here (eg. `if (==null) return error; return applyAndEnqueue...`), mostly since the `applyAndEnqueue` expression is non-trivial. Is there any official style guidance on this?

Reply:
the check was actually just redundant, so it's gone.